components/common/src/templating/config.rs

```diff
@@ -587,35 +573,109 @@ fn is_toml_value_a_table(key: &str, table: &toml::value::Table) -> bool {
     }
 }
 
-#[cfg(not(windows))]
-fn set_permissions<T: AsRef<Path>>(path: T, pkg: &Pkg) -> hcore::error::Result<()> {
+#[cfg(unix)]
+fn set_permissions(path: &Path, user: &str, group: &str) -> hcore::error::Result<()> {
     use crate::hcore::os::users;
     use crate::hcore::util::posix_perm;
 
     if users::can_run_services_as_svc_user() {
-        posix_perm::set_owner(path.as_ref(), &pkg.svc_user, &pkg.svc_group)?;
+        posix_perm::set_owner(path, &user, &group)?;
     }
-    posix_perm::set_permissions(path.as_ref(), CONFIG_PERMISSIONS)
+
+    let permissions = if path.is_dir() {
+        CONFIG_DIR_PERMISSIONS
+    } else {
+        CONFIG_PERMISSIONS
+    };
+    posix_perm::set_permissions(&path, permissions)
 }
 
 #[cfg(windows)]
-fn set_permissions<T: AsRef<Path>>(path: T, _pkg: &Pkg) -> hcore::error::Result<()> {
+fn set_permissions(path: &Path, _user: &str, _group: &str) -> hcore::error::Result<()> {
     use crate::hcore::util::win_perm;
 
-    win_perm::harden_path(path.as_ref())
+    win_perm::harden_path(&path)
```

Comment:
This is already a reference
```suggestion
    win_perm::harden_path(path)
```